x-pack/plugins/observability/public/pages/cases/helpers.ts

```diff
@@ -23,31 +105,35 @@ export const useFetchAlertDetail = (alertId: string): [boolean, TopAlert | null]
 
   useEffect(() => {
     const abortCtrl = new AbortController();
+
     const fetchData = async () => {
       try {
         setLoading(true);
-        const response = await http.get<Record<string, unknown>>('/internal/rac/alerts', {
+        const response = await http.get<Record<string, unknown>>(BASE_RAC_ALERTS_API_PATH, {
           query: {
             id: alertId,
           },
+          signal: abortCtrl.signal,
         });
         if (response) {
           const parsedAlert = parseAlert(observabilityRuleTypeRegistry)(response);
+
           setAlert(parsedAlert);
-          setLoading(false);
```

Comment:
I think we want to set loading to false when we're done, not only when we successfully get data?